src/components/WorkOrders/UpdateJob.js

```diff
@@ -0,0 +1,145 @@
+import PropTypes from 'prop-types'
+import { useState, useEffect } from 'react'
+import Spinner from '../Spinner/Spinner'
+import BackButton from '../Layout/BackButton/BackButton'
+import ErrorMessage from '../Errors/ErrorMessage/ErrorMessage'
+import { getTask } from '../../utils/frontend-api-client/tasks'
+import { getSorCodes } from '../../utils/frontend-api-client/schedule-of-rates/codes'
+import UpdateJobForm from './UpdateJobForm'
+import SummaryUpdateJob from './SummaryUpdateJob'
+import { updatedTasks } from '../../utils/update-job'
+import { buildUpdateJob } from '../../utils/hact/update-job'
+import { postUpdateJob } from '../../utils/frontend-api-client/work-orders'
+import { useRouter } from 'next/router'
+
+const CloseJob = ({ reference }) => {
+  const [task, setTask] = useState([])
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState()
+  const [sorCodes, setSorCodes] = useState([])
+  const [sorCodesCollection, setSorCodesCollection] = useState([])
+  const [showSummaryPage, setShowSummaryPage] = useState(false)
+  const [showAddedSoreCodes, setShowAddedSoreCodes] = useState(false)
+  const router = useRouter()
+
+  const onGetToSummary = (e) => {
+    const tasks = updatedTasks(e)
+    setSorCodesCollection(
+      e.sorCodesCollection
+        ? e.sorCodesCollection
+            .filter((e) => e != null)
+            .map((e, index) => {
+              return { id: index, ...e }
+            })
+        : []
+    )
+
+    setTask(tasks)
+    setShowSummaryPage(true)
+  }
+
+  const changeCurrentPage = () => {
+    setShowAddedSoreCodes(true)
+    setShowSummaryPage(false)
+  }
+
+  const onJobUpdateSubmit = () => {
+    const updateJobFormData = buildUpdateJob(
+      task,
+      sorCodesCollection,
+      reference
+    )
+    makePostRequest(updateJobFormData)
+  }
+
+  const makePostRequest = async (formData) => {
+    setLoading(true)
+
+    try {
+      await postUpdateJob(formData)
+      router.push('/')
```

Comment:
what is this doing ?

Reply:
it redirects contractors to work-orders list table